### DISCUSSION TITLE
Is the Grid Geo Sampler sampled by random?

### DISCUSSION THREAD

**Discussion by Raji-Ahmad** [torchgeo/torchgeo#1020] [Q&A]
I need to sample grids of N X N pixels from the train dataset with all pixels included in the sampler. I know the Grid Geo Sampler can achieve this and the stride functionality is a great option to create more samples. 

My issue is that some features of interest might be clustered at some part of the dataset and possibly make only  some successive batch have this feature if it's not sampled by random.

Does the Grid Geo Sampler sample the grids by random and if not how can I sample all grids by random, more like a Random Grid Geo Sampler?

**Comment by adamjstewart** [comment]
GridGeoSampler does not sample grids by random, and we don't yet have a RandomGridGeoSampler. You may also be interested in #757.
